### DISCUSSION TITLE
Searching for the RSS feed of a website

### DISCUSSION THREAD

**Discussion by autay27** [spacecowboy/Feeder#319] [Ideas]
It would be nice if I could enter the URL of a website and be given the option to subscribe to its RSS feed, even if it is served at a subpage. Sometimes the RSS feed is not explicitly linked anywhere on a page and it's a pain to search for it manually. 

The app just needs to search a few common URL suffixes such as /feed.xml, /feed, /rss... 

An example implementation is [this website](https://www.rsslookup.com/) whose source code is available [here](https://github.com/mratmeyer/rsslookup), I believe the logic is [here](https://github.com/mratmeyer/rsslookup/blob/main/worker/index.js#L7).

**Comment by spacecowboy** [comment]
Feeder already does this.

Foe example, try inputting just "theverge.com"

and Feeder will find the feed.

It works if the site has included metadata where the feed is located.

**Reply by autay27** [reply]
Yes, the existing capability is good. But I think it would be nice if it does this even when the metadata is not included. 

There are many such cases, including those with no links e.g. ([x](https://austinkleon.com/)) or those which have the RSS feed linked somewhere but not the metadata e.g. ([x](https://tratt.net/laurie/)). 

If the idea sounds sensible I can try to work on it.

**Reply by markcellus** [reply]
As a website owner that provides RSS feeds, I would not want an app going through all sorts of non-existent paths on my domain looking for rss feeds that arent there. It's a unnecessary use of a server's resources and can lead to the requests being blocked.

If this is implemented, I'd be very careful, that it doesn't do more harm than good.

**Reply by spacecowboy** [reply]
indeed. I've heard from more than one website owner about too many requests. 

people should learn to use established metadata conventions if they want users to find the feed

**Reply by autay27** [reply]
All right, thank you for explaining.

I am surprised that just ~5 requests to different paths on a website would be registered as suspicious. I can open 5 different tabs of a website easily while exploring it. I guess it is different because it comes from an app rather than a browser? Or is it because the paths are not (necessarily) real pages on the domain?

**Reply by spacecowboy** [reply]
One website owner contacted me after feeder made 2! requests too quickly to the server.
